test/integration/targets/consul/tasks/consul_session.yml

```diff
@@ -82,6 +82,51 @@
     that:
       - result is failed
 
+- when: pyopenssl_version.stdout is version('0.15', '>=')
+  block:
+    - name: ensure SSL certificate is checked
+      consul_session:
+        state: info
+        id: '{{ session_id }}'
+        port: 8501
+        scheme: https
+      register: result
+      ignore_errors: True
+
+    - name: previous task should fail since certificate is not known
+      assert:
+        that: result is failed
```

Comment:
Maybe add a check that `result.msg` contains something indicating that certificate check is actually failing.

Reply:
Updated